src/resmom/catch_child.c

```diff
@@ -1888,53 +1888,90 @@ scan_for_exiting(void)
 }
 
 /**
- * @brief	Send a restart message to the Server over IS protocol
+ * @brief
+ * 		choosing one server in random if a failover server is already set up.
+ *
+ * @param[out] port - Passed through to parse_servername(), not modified here.
+ *
+ * @return char *
+ * @return NULL - failure
+ * @retval !NULL - pointer to server name
+ */
+char *
+get_servername_random(unsigned int *port)
```

Comment:
how will this work? If both primary and failover are up, then we must connect to primary right? Because secondary will be inactive

Reply:
So secondary will reject the request as it could not obtain the node related information.

Reply:
secondary may not be listening at all right? what happens then? did you test with this configuration?

Reply:
You are right. Comm is rejecting the message saying dest not found.

```
PBS_PRIMARY=centos7
PBS_SECONDARY=centos7_2


Mom Log:
06/08/2020 14:39:12;0006;pbs_mom;Fil;pbs_mom;Version 20.0.0, started, initialization type = 0
06/08/2020 14:39:12;0002;pbs_mom;Svr;pbs_mom;Mom pid = 5424 ready, using ports Server:15001 MOM:15002 RM:15003
06/08/2020 14:39:12;0002;pbs_mom;Svr;pbs_mom;HELLO sent to server at centos7_2:15001
06/08/2020 14:39:12;0d80;pbs_mom;TPP;pbs_mom(Thread 0);sd 0, Received noroute to dest 172.17.0.5:15001, msg="tfd=16, pbs_comm:172.17.0.4:17001: Dest not found"
06/08/2020 14:39:12;0d80;pbs_mom;TPP;pbs_mom(Main Thread);net restore handler called
06/08/2020 14:39:12;0001;pbs_mom;Svr;pbs_mom;im_eof, Premature end of message from addr 172.17.0.5:15001 on stream 0
06/08/2020 14:39:12;0002;pbs_mom;Svr;im_eof;Server closed connection.

06/08/2020 14:39:14;0002;pbs_mom;Svr;pbs_mom;HELLO sent to server at centos7_2:15001
06/08/2020 14:39:14;0d80;pbs_mom;TPP;pbs_mom(Thread 0);sd 1, Received noroute to dest 172.17.0.5:15001, msg="tfd=16, pbs_comm:172.17.0.4:17001: Dest not found"
06/08/2020 14:39:14;0001;pbs_mom;Svr;pbs_mom;im_eof, Premature end of message from addr 172.17.0.5:15001 on stream 1
06/08/2020 14:39:14;0002;pbs_mom;Svr;im_eof;Server closed connection.
06/08/2020 14:39:18;0002;pbs_mom;Svr;pbs_mom;HELLO sent to server at centos7_2:15001
06/08/2020 14:39:18;0d80;pbs_mom;TPP;pbs_mom(Thread 0);sd 2, Received noroute to dest 172.17.0.5:15001, msg="tfd=16, pbs_comm:172.17.0.4:17001: Dest not found"
06/08/2020 14:39:18;0001;pbs_mom;Svr;pbs_mom;im_eof, Premature end of message from addr 172.17.0.5:15001 on stream 2
06/08/2020 14:39:18;0002;pbs_mom;Svr;im_eof;Server closed connection.
06/08/2020 14:39:22;0002;pbs_mom;Svr;pbs_mom;HELLO sent to server at centos7:15001
06/08/2020 14:39:22;0002;pbs_mom;Svr;pbs_mom;ReplyHello from server at 172.17.0.4:15001
```